vms/platformvm/txs/mempool/mempool.go

```diff
@@ -199,20 +215,32 @@ func (m *mempool) Remove(txsToRemove []*txs.Tx) {
 }
 
 func (m *mempool) Peek() (*txs.Tx, bool) {
+	m.lock.RLock()
```

Comment:
the lock doesn't need to be held here

Reply:
This is needed so that during the course of iteration the map is not modified from a concurrent call to `Add` ([ref](https://github.com/ava-labs/avalanchego/blob/f0a86cc86edb5ac95095063fa1c53e6f5de652bf/utils/linkedhashmap/iterator.go#L16-L18)).

Reply:
Strange- I thought I commented this on `Peek`. We definitely need the lock on `Iterate`, sorry!